src/System.Diagnostics.TraceSource/src/System/Diagnostics/CorrelationManager.cs

```diff
@@ -33,7 +34,13 @@ public Stack LogicalOperationStack
         {
             get
             {
-                return GetLogicalOperationStack();
+                ImmutableArray<object> elements = _idStack.Value.ToImmutableArray();
+                Stack newStack = new Stack();
```

Comment:
Nit: Specify the capacity to avoid intermediate allocations as you're pushing elements. e.g.:

```c#
Stack newStack = new Stack(elements.Length);
```